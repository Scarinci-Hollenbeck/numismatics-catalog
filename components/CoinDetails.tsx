import React from 'react';
import BreadCrumbs from './BreadCrumbs';
import CoinImage from './CoinImage';

export default function CoinDetails({ coin, crumbs }): JSX.Element {
  return (
    <>
      <BreadCrumbs crumbs={crumbs} />
      <section>
        <CoinImage image={coin.imageUrl} title={coin.title} />
        <h2>{coin.title}</h2>
        <p>{coin.description}</p>
      </section>

      <style jsx>
        {`
          section {     
            background-color: #fff;
            box-shadow: 2px 4px 15px black;
            max-width: 100%;
            padding: 1em;
            border-radius:37px;
            margin-bottom: 3em;
            font-family: 'Tajawal Regular';
            font-size: 1.3rem; 
            text-align: center;
          }

          p {
            margin-top: 0;
          }
        `}
      </style>
    </>
  );
}
