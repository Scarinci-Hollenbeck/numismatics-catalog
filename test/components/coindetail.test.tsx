import React from 'react';
import renderer from 'react-test-renderer';
import CoinDetails from '../../components/CoinDetails';

const coins = [
  {
    title: 'King Alfonso XII, 1874 - 1885',
    description:
      'Obv: Alfonso XII Por La G De Dios 1874-1885, Rev: El Pacificador Barcelona MCMLXII',
    category: 'Calico Kings and Queens of Spain',
    imageUrl:
      'http://res.cloudinary.com/numismatics-catalog/image/upload/v1601061183/coins/kw2osskjam04mvgiokef.jpg',
    categoryId: '5f6e4147ae97b21b48da848e',
    width: 500,
    height: 250,
  },
  {
    title: 'King Alfonso XII, 1874 - 1885',
    description:
      'Obv: Alfonso XII Por La G De Dios 1874-1885, Rev: El Pacificador Barcelona MCMLXII',
    category: 'Calico Kings and Queens of Spain',
    imageUrl:
      'http://res.cloudinary.com/numismatics-catalog/image/upload/v1601061183/coins/kw2osskjam04mvgiokef.jpg',
    categoryId: '5f6e4147ae97b21b48da848e',
    width: 500,
    height: 250,
  },
];

const crumbs = [
  {
    link:
      'http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xii%2C-1874---1885',
    title: 'Coin Item 1',
  },
  {
    link:
      'http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xi%2C-1840---1874',
    title: 'Coin Item 2',
  },
];

describe('Coin Detail Component', () => {
  it('expect coins length == 2', () => {
    expect(coins.length).toBe(2);
  });

  it('The word "King" is found in the title', () => {
    const { title } = coins[0];
    expect(title).toMatch(/King/);
  });

  it('the coin article list contains sourceUrl and title', () => {
    expect(coins[1]).toHaveProperty('title', 'King Alfonso XII, 1874 - 1885');
    expect(coins[1]).toHaveProperty(
      'imageUrl',
      'http://res.cloudinary.com/numismatics-catalog/image/upload/v1601061183/coins/kw2osskjam04mvgiokef.jpg',
    );
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(<CoinDetails coin={coins[0]} crumbs={crumbs} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
