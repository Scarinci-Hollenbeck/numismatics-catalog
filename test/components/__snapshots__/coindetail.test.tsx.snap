// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin Detail Component matches snapshot 1`] = `
Array [
  <ul
    className="jsx-3026771491 breadcrumbs"
  >
    <li
      className="jsx-3026771491"
    >
      <a
        className="jsx-3026771491"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="jsx-3026771491"
    >
      <a
        className="jsx-3026771491"
        href="http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xii%2C-1874---1885"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Coin Item 1
      </a>
    </li>
    <li
      className="jsx-3026771491"
    >
      <a
        className="jsx-3026771491"
        href="http://test.com/coin/5f6e4147ae97b21b48da848e/king-alfonso-xi%2C-1840---1874"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Coin Item 2
      </a>
    </li>
  </ul>,
  <section
    className="jsx-752553799"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjI1MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "border": "none",
              "display": "block",
              "margin": 0,
              "maxWidth": "100%",
              "padding": 0,
            }
          }
        />
      </div>
      <img
        alt="King Alfonso XII, 1874 - 1885"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
    <h2
      className="jsx-752553799"
    >
      King Alfonso XII, 1874 - 1885
    </h2>
    <p
      className="jsx-752553799"
    >
      Obv: Alfonso XII Por La G De Dios 1874-1885, Rev: El Pacificador Barcelona MCMLXII
    </p>
  </section>,
]
`;
